### PR TITLE
Added two new functions to Microbiome Modeling Toolbox

### PR DESCRIPTION
**Pull request by almut-heinken**
-Function to create pan-models from AGORA models
-Function to extract shadow prices from solution
-Also reorganized folder

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

### CODE REVIEW

File: src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+% are stratifying the modeled personalized microbiomes. The fecal exchanges 
+% secreting the metabolites of interest (e.g., EX_co2[fe]) should be used 
+% as the objective functions entered in the present function to determine
+% model compounds that have value for the secretion of the metabolite of
+% interest.
+%
+% USAGE:
+% [ShadowPrices]=analyseObjectiveShadowPrices(modelList,objectiveList,varargin)
+%
+% REQUIRED INPUTS:
```
Comment (laurentheirendt) [inline src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m]: This function header needs proper formatting.

File: src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    end
+else
+    % without parallelization
+    for i=1:size(modelList,1)
+        ShadowPrices{1,i+2}=modelIDs{i,1};
+        model=modelList{i,1};
+        % Compute the solutions for all objectives
+        for j=1:size(objectiveList,1)
+            % optimize for the objective if it is present in the model
+            if ~isempty(find(ismember(model.rxns,objectiveList{j,1})))
```
Comment (laurentheirendt) [inline src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m]: May you use here an internal function?

File: src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+extractedShadowPrices={};
+% Find all shadow prices (negative or positive depending on variable
+% SPDef)
+cnt=1;
+for i=1:length(model.mets)
+    % Do not include slack variables
+    if ~strncmp('slack_',model.mets{i},6)
+        if strcmp(SPDef,'Negative')
+            if FBAsolution.dual(i)  <0 && abs(FBAsolution.dual(i))>0.00000001
```
Comment (laurentheirendt) [inline src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m]: May you use here a `tol` variable and set it with scientific notation?

File: src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/createPanModels.m [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+% be combined into a pan-reconstruction. The pan-biomass reactions will be
+% built from the average of all biomasses. Futile cycles that result from
+% the newly combined reaction content are removed by setting certain
+% reactions irreversible. These reactions have been determined manually.
+% NOTE: Futile cycle removal has only been tested at the species and genus
+% level. Pan-models at higher taxonomical levels (e.g., family) may
+% contain futile cycles and produce unrealistically high ATP flux.
+% The pan-models can be used an input for mgPipe if taxon abundance data is
+% available at a higher level than strain, e.g., species, genus.
+%
```
Comment (laurentheirendt) [inline src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/createPanModels.m]: This function header needs proper formatting.

File: src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/createPanModels.m [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
... PGK AND GAPD_NADP AND GAPD','G1PPT','4 h[c] + 2 g1p[c] <=> glc_D[c] + M01966[c]','G1PPTi','4 h[c] + 2 g1p[c] -> glc_D[c] + M01966[c]';'PPIt2e AND GUAPRT AND AACPS6 AND GALT','PPIt2e','h[c] + ppi[c] -> h[e] + ppi[e]','PPIte','ppi[c]  -> ppi[e]';'PPIt2e AND GLGC AND NADS2 AND SADT','PPIt2e','h[c] + ppi[c] -> h[e] + ppi[e]','PPIte','ppi[c]  -> ppi[e]';'MCOATA AND MALCOAPYRCT AND C180SNrev','MCOATA','ACP[c] + malcoa[c]  <=> coa[c] + malACP[c]','MACPMT','ACP[c] + malcoa[c] -> coa[c] + malACP[c]';'PPCr AND MALCOAPYRCT AND MMSAD5 AND MSAS','PPCr','h[c] + pi[c] + oaa[c] <=> h2o[c] + co2[c] + pep[c]','PPC','co2[c] + h2o[c] + pep[c] -> h[c] + oaa[c] + pi[c]';'ACt2r AND ACtr','ACtr','ac[e]  <=> ac[c]',[],[];'LEUt2r AND LEUtec','LEUtec','leu_L[e]  <=> leu_L[c]',[],[];'PTRCt2r AND PTRCtex2','PTRCtex2','ptrc[c]  <=> ptrc[e]',[],[];'TYRt2r AND TYRt','TYRt','tyr_L[e]  <=> tyr_L[c] ',[],[];'OCBT AND CITRH AND CBMKr','CBMKr','nh4[c] + atp[c] + co2[c]  <=> 2 h[c] + adp[c] + cbp[c]','CBMK','nh4[c] + atp[c] + co2[c]  -> 2 h[c] + adp[c] + cbp[c]';'TSULt2 AND SO3t AND H2St AND TRDRr','TRDRr','h[c] + nadph[c] + trdox[c]  <=> nadp[c] + trdrd[c] ','TRDR','h[c] + nadph[c] + trdox[c] -> nadp[c] + trdrd[c]'};
```
Comment (laurentheirendt) [inline src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/createPanModels.m]: Can this be included in a separate file?

File: src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/createPanModels.m [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    model=readCbModel([panPath,panModels{i}]);
+    model=useDiet(model,dietConstraints);
+    model=changeObjective(model,'DM_atp_c_');
+    FBA=optimizeCbModel(model,'max');
+    % Ensure that pan-models can still produce biomass
+    model=changeObjective(model,'biomassPan');
+    if FBA.f > 50
+        for j=2:size(reactionsToReplace,1)
+            rxns=strsplit(reactionsToReplace{j,1},' AND ');
+            go=1;
```
Comment (laurentheirendt) [inline src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/createPanModels.m]: The variable `go` is boolean. May you use `go = true`?

**Comment by almut-heinken** [comment]
Hi @laurentheirendt 
I will make the proposed changes.
Best

File: src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m (hunk 2 above)
Comment (almut-heinken) [reply src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m]: @laurentheirendt I don't understand the comment.

File: src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m (hunk 3 above)
Comment (almut-heinken) [reply src/analysis/multiSpecies/microbiomeModelingToolbox/additionalAnalysis/analyseObjectiveShadowPrices.m]: @laurentheirendt I am not sure what this means either.

**Comment by laurentheirendt** [comment]
Thanks @almut-heinken 👍

**Pull request merged**